auth/jwt_test.go

```diff
@@ -47,48 +50,134 @@ func (suite *JWTAuthenticatorTestSuite) createWrongToken(method jwt.SigningMetho
 	return token.SignedString([]byte(config.GetString("auth.jwt.secret")))
 }
 
-func (suite *JWTAuthenticatorTestSuite) TestAuthenticate() {
-	user := &TestUser{}
-	tokenAuthenticator := &JWTAuthenticator{}
+func (suite *JWTAuthenticatorTestSuite) TestGenerateToken() {
 	token, err := GenerateToken("johndoe@example.org")
 	suite.Nil(err)
-	suite.Nil(tokenAuthenticator.Authenticate(suite.createRequest(token), user))
-	suite.Equal("Admin", user.Name)
+	suite.Nil(suite.tokenAuthenticator.Authenticate(suite.createRequest(token), suite.user))
+	suite.Equal("Admin", suite.user.Name)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestGenerateTokenValidates() {
+	token, err := GenerateToken("johndoe@example.org")
+	suite.Nil(err)
+	claims := jwt.MapClaims{}
+	_, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(config.GetString("auth.jwt.secret")), nil
+	})
+	suite.Nil(err)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestGenerateTokenHasClaims() {
+	token, err := GenerateToken("johndoe@example.org")
+	suite.Nil(err)
+	claims := jwt.MapClaims{}
+	_, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(config.GetString("auth.jwt.secret")), nil
+	})
+	suite.Nil(err)
+
+	useridPresent := false
+	for key, val := range claims {
+		if key == "userid" {
+			suite.Equal("johndoe@example.org", val)
+			useridPresent = true
+		}
+	}
+
+	suite.True(useridPresent)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestGenerateTokenWithClaimsHasClaims() {
+	token, err := GenerateTokenWithClaims("johndoe@example.org", jwt.MapClaims{
+		"sub": suite.user.ID,
+	})
+	suite.Nil(err)
+	claims := jwt.MapClaims{}
+	_, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(config.GetString("auth.jwt.secret")), nil
+	})
+	suite.Nil(err)
 
-	user = &TestUser{}
-	token, err = GenerateToken("wrongemail@example.org")
+	useridPresent := false
+	subPresent := false
+	for key, val := range claims {
+		if key == "userid" {
+			suite.Equal("johndoe@example.org", val)
+			useridPresent = true
+		}
+		if key == "sub" {
+			suite.Equal(suite.user.ID, uint(val.(float64)))
+			subPresent = true
+		}
+	}
+	suite.True(useridPresent)
+	suite.True(subPresent)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestGenerateTokenWithClaimsValidates() {
```

Comment:
This test is redundant with `TestGenerateTokenWithClaimsHasClaims()`  (and tests `GenerateToken` instand of `GenerateTokenWithClaims`